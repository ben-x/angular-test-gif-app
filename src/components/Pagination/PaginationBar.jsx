import React, {useState} from 'react';


const PaginationBar = ({limit, page, pagination, setPage}) => {
   const {total_count, count} = pagination
   const noOfPages = Math.ceil(total_count/count)
   const [val, setVal] = useState({val1:0, val2:10})
   const {val1, val2} = val;
   const getNumbers = () => {
      let pageNumbers = [];
      for(let i=1; i<= noOfPages; i++) {
         pageNumbers.push(`${i}`);
       }

       pageNumbers = pageNumbers.slice(val1, val2)
       return  pageNumbers.map(number => {
         return (
           <i className={ page === number ? "activePage" : "pagesNumbered"} onClick={() => setPage(number)} >
              {number}
           </i>
         );
       })
   }
   return (
      <div style={{ margin:'20px 20px 70px 20px' }}>
         <p></p>
       { val1 === 0 ? '' : (
          <>
       <i
         className="fa fa-arrow-left"
         aria-hidden="true" >
         </i> <span style={{ cursor: 'pointer' }} onClick={()=>setVal({val1: val1 === 1 ? val1 : val1-1, val2:val2 === 10 ? val2 : val2-1})}>Prev</span> 
         </>
          )
      } 
         {
           getNumbers()
         }

         { page >= noOfPages ? '' : ( <>  <i
         className="fa fa-arrow-right"
         aria-hidden="true"
         ></i><span style={{ cursor: 'pointer' }} onClick={()=>setVal({val1:val1+1, val2:val2+1})}>Next</span> 
         </>
         )
}
      </div>
    )
}

export default PaginationBar;
