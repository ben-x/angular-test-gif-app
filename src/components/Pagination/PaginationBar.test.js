import React from "react";
import { shallow } from "enzyme";
import PaginationBar from "./PaginationBar";

const props = {
    limit: 25,
    page: 1,
    pagination: {},
    setPage: jest.fn()
};

const wrap = shallow(<PaginationBar {...props} />);

describe("<PaginationBar />", () => {
    it("renders the component successfully", () => {
        expect(wrap).toMatchSnapshot();
    });
});
