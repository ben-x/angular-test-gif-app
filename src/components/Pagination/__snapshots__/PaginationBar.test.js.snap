// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationBar /> renders the component successfully 1`] = `ShallowWrapper {}`;
